packages/core/upload/admin/src/components/TableList/index.js

```diff
@@ -40,12 +47,55 @@ export const TableList = ({
               }
             />
           </Th>
-          {tableHeaders.map(({ metadatas, key }) => {
+          {tableHeaders.map(({ metadatas: { label, isSortable }, name, key }) => {
+            const isSorted = sortBy === name;
+            const isUp = sortOrder === 'ASC';
+            const tableHeaderLabel = formatMessage(label);
+            const sortLabel = formatMessage(
+              { id: 'list-table-header-sort', defaultMessage: 'Sort on {label}' },
+              { label: tableHeaderLabel }
+            );
+
+            const handleClickSort = () => {
```

Comment:
Is it worth defining this as a parameterised function outside the map so it's not created each time? Something like:

```
  const handleSortClick = ({ isSortable = false, isSorted = false, name }) => {
    if (!isSortable || !name) {
      return;
    }
    const nextSortOrder = isSorted && sortOrder === 'ASC' ? 'DESC' : 'ASC';
    const nextSort = `${name}:${nextSortOrder}`;

    onChangeSort(nextSort);
  };
```

Reply:
that sounds like a good idea indeed, thanks!